content/en/docs/concepts/configuration/secret.md

```diff
@@ -911,20 +926,14 @@ data:
   extra: YmFyCg==
 ```
 
-When creating a `Pod`, Kubernetes automatically finds or creates a service account
-Secret and then automatically modifies your Pod to use this Secret. The service account
-token Secret contains credentials for accessing the Kubernetes API.
-
-The automatic creation and use of API credentials can be disabled or
-overridden if desired. However, if all you need to do is securely access the
-API server, this is the recommended workflow.
+After creating the Secret, wait for the `token` key in the `data` field to be populated.
```

Comment:
Out of curiosity, due to my other work in recognizing reconciliation: Does the secret have a ready condition or at least observedGeneration that gets set after the token is populated by the controller?

Reply:
No, you can tell the token is populated because the token field is set ;-)

There's no status, conditions, or observedGeneration fields in the Secret API

Reply:
That’s unfortunate. I’m gonna have to code an exception handler for Secrets that checks for the annotation to know to wait for the token before it’s considered reconciled.

Reply:
Not sure I understand what you mean… the auto-generated secrets have the annotation as well. The thing to wait for is the presence of the token

Reply:
cli-utils wasn't waiting for the Secret tokens before, probably because no one was applying them directly. So no one filed a bug report or feature request.

When a Secret is auto-generated from a ServiceAccount, the applier doesn't know that (without special business logic). It just knows to apply the ServiceAccount and wait for the ServiceAccount to be ready. But the ServiceAccount doesn't have status either. So it was considered ready immediately (which I now know is technically incorrect, because the controller takes action after seeing an update).

In legacy mode, is there any indication on the ServiceAccount that the Secret has been successfully generated?

Reply:
> In legacy mode, is there any indication on the ServiceAccount that the Secret has been successfully generated?

The names of any autogenerated token secrets were added to the serviceaccount's `.secrets` list

Reply:
Of course, duh. Thanks. I'll need to add that to kstatus as an exception too.

Reply:
Actually, I can't now, without knowing if the feature flag is enabled... Is there any way to look that up programmatically from the API?

Reply:
This is where a status condition would be helpful. Then it could be marked as ready either way, by the controller.

Reply:
Let sync up offline… I'm not sure kstatus (or anything else) should be waiting on the .secrets field… I'm not quite sure what you're trying to accomplish there.